app/services/service_provider_updater.rb

```diff
@@ -20,11 +20,24 @@ def update_local_caches(service_provider)
 
   def update_cache(issuer, service_provider)
     if service_provider['active'] == true
-      ServiceProvider.find_or_create_by!(issuer: issuer) do |sp|
-        sp.attributes = cleaned_service_provider(service_provider)
-      end
+      create_or_update_service_provider(issuer, service_provider)
     else
-      ServiceProvider.destroy_all(issuer: issuer, approved: false)
+      ServiceProvider.destroy_all(issuer: issuer, native: false)
+    end
+  end
+
+  def create_or_update_service_provider(issuer, service_provider)
+    sp = ServiceProvider.from_issuer(issuer)
+    return if sp.native?
+    sync_model(sp, cleaned_service_provider(service_provider))
+  end
+
+  def sync_model(sp, cleaned_attributes)
+    if sp.is_a?(NullServiceProvider)
```

Comment:
@jessieay valid question.

We decided on the current functionality because throughout the app we had lots of existing code that tried to initiate a `ServiceProvider` from an issuer string and then act on the resulting object. We didn't want new `ServiceProviders` to be written to the db as an accidental side effect of initializing one from an issuer string.

Here we *do* want to write to the db, hence the extra check. Seems like we are erring on the side of caution.